machine/v1beta1/types_azureprovider.go

```diff
@@ -34,6 +34,9 @@ type AzureMachineProviderSpec struct {
 	Image Image `json:"image"`
 	// OSDisk represents the parameters for creating the OS disk.
 	OSDisk OSDisk `json:"osDisk"`
+	// DataDisk specifies the parameters that are used to add one or more data disks to the machine
```

Comment:
What gets placed on data disk versus an os disk?

Reply:
OS Disk
_"Every virtual machine has one attached operating system disk. That OS disk has a pre-installed OS, which was selected when the VM was created. This disk contains the boot volume.
This disk has a maximum capacity of 4,095 GiB, however, many operating systems are partitioned with master boot record MBR by default. MBR limits the usable size to 2 TiB. If you need more than 2 TiB, create and attach [data disks](https://docs.microsoft.com/en-us/azure/virtual-machines/managed-disks-overview#data-disk) and use them for data storage"_

Data Disk
_"A data disk is a managed disk that's attached to a virtual machine to store application data, or other data you need to keep. Data disks are registered as SCSI drives and are labeled with a letter that you choose. Each data disk has a maximum capacity of 32,767 gibibytes"_

Also: _"Ultra Disks can't be used as OS Disks, they can only be created as empty Data Disks"_

Would you update the godoc comment to explain some if this in there?

Reply:
Additional information in the godoc for this field would be useful. I think telling the user why they might want a data disk or how they might use it, (I guess via a host volume?), would be useful